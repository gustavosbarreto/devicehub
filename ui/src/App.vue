<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
