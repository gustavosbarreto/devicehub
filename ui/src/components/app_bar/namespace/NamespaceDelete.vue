<template>
  <fragment>
    <v-btn
      color="red darken-1"
      outlined
      data-test="delete-btn"
      @click="dialog = !dialog"
    >
      Delete namespace
    </v-btn>

    <v-dialog
      v-model="dialog"
      max-width="400"
    >
      <v-card data-test="namespaceDelete-dialog">
        <v-card-title class="headline grey lighten-2 text-center">
          Are you sure?
        </v-card-title>

        <v-card-text class="mt-4 mb-3 pb-1">
          <p
            v-if="active && amountDue !== null"
            data-test="contentSubscription-p"
          >
            Deleting the namespace will generate an invoice,
            estimated <b> {{ amountDue | formatCurrency }} </b> for the time of use.
          </p>

          <p data-test="content-text">
            This action cannot be undone. This will permanently delete the
            <b> {{ displayOnlyTenCharacters(name) }} </b>and its related data.
          </p>
        </v-card-text>

        <v-card-actions>
          <v-spacer />

          <v-btn
            text
            data-test="close-btn"
            @click="dialog=!dialog"
          >
            Close
          </v-btn>

          <v-btn
            color="red darken-1"
            text
            data-test="remove-btn"
            @click="remove();"
          >
            Remove
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </fragment>
</template>

<script>

import formatCurrency from '@/components/filter/currency';

export default {
  name: 'NamespaceDelete',

  filters: { formatCurrency },

  props: {
    nsTenant: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      name: '',
      dialog: false,
      card: null,
      elements: null,
      amountDue: null,
      cardItems: {},
    };
  },

  computed: {
    tenant() {
      return this.$props.nsTenant;
    },

    active() {
      return this.$store.getters['billing/active'];
    },

    billing() {
      return this.$store.getters['billing/get'];
    },

    isOwner() {
      return this.$store.getters['namespaces/owner'];
    },
  },

  async created() {
    this.name = this.$store.getters['namespaces/get'].name;
  },

  mounted() {
    if (this.isOwner && this.isBillingEnabled()) {
      this.getSubscriptionInfo();
    }
  },

  methods: {
    isBillingEnabled() {
      return this.$env.billingEnable;
    },

    async remove() {
      try {
        this.dialog = !this.dialog;
        await this.$store.dispatch('namespaces/remove', this.tenant);
        await this.$store.dispatch('auth/logout');
        await this.$router.push({ name: 'login' });
        this.$store.dispatch('snackbar/showSnackbarSuccessAction', this.$success.namespaceDelete);
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.snackbar.namespaceDelete);
      }
    },

    displayOnlyTenCharacters(str) {
      if (str !== undefined) {
        if (str.length > 10) return `${str.substr(0, 10)}...`;
      }
      return str;
    },

    getDueAmount(data) {
      return data.upcoming_invoice.amount_due;
    },

    async getSubscriptionInfo() {
      if (this.active) {
        try {
          const data = await this.$store.dispatch('billing/getSubscription');
          this.amountDue = this.getDueAmount(data);
        } catch {
          this.$store.dispatch('snackbar/showSnackbarErrorDefault');
        }
      }
    },
  },
};

</script>
