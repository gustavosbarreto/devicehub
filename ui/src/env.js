export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
