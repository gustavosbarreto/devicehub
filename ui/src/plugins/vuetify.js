import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import '@mdi/font/css/materialdesignicons.css';
import 'font-logos/assets/font-logos.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { fas } from '@fortawesome/free-solid-svg-icons';
import Clipboard from 'v-clipboard';
import Fragment from 'vue-fragment';

//import 'vuetify/dist/vuetify.min.css'
//import '../styles/variables.scss'

Vue.component('font-awesome-icon', FontAwesomeIcon); // Register component globally
library.add(fas); // Include needed icons.

Vue.use(Vuetify);
Vue.use(Clipboard);
Vue.use(Fragment.Plugin);

const DEVICE_ICONS = {
  alpine: 'fl-alpine',
  arch: 'fl-archlinux',
  centos: 'fl-centos',
  coreos: 'fl-coreos',
  debian: 'fl-debian',
  devuan: 'fl-devuan',
  elementary: 'fl-elementary',
  fedora: 'fl-fedora',
  freebsd: 'fl-freebsd',
  gentoo: 'fl-gentoo',
  linuxmint: 'fl-linuxmint',
  mageia: 'fl-mageia',
  manjaro: 'fl-manjaro',
  mandriva: 'fl-mandriva',
  nixos: 'fl-nixos',
  opensuse: 'fl-opensuse',
  rhel: 'fl-redhat',
  sabayon: 'fl-sabayon',
  slackware: 'fl-slackware',
  ubuntu: 'fl-ubuntu',
  raspbian: 'fl-raspberry-pi',
  'ubuntu-core': 'fl-ubuntu',
  void: 'fl-void',
};

export default new Vuetify({
  iconfont: 'md',
  icons:{
    values:DEVICE_ICONS
  },
  theme: {
    dark: false,
    themes: {
      light: {
        primary: '#364576',
        secondary: '#b0bec5',
        accent: '#8c9eff',
        error: '#b71c1c',
      },
    }
  }
});
