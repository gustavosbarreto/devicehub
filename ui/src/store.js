import Vue from 'vue';
import Vuex from 'vuex';
import stats from '@/modules/stats';
import sessions from '@/modules/sessions';
import auth from '@/modules/auth';
import devices from '@/modules/devices';
import modals from '@/modules/modals';
import firewallrules from '@/modules/firewall_rules';
import notifications from '@/modules/notifications';

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    devices, modals, stats, sessions, auth, firewallrules, notifications,
  },
});
