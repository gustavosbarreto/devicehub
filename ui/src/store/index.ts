import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { apiKeys, ApiKeysState } from "./modules/api_keys";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { customer, CustomerState } from "./modules/customer";
import { announcement, AnnouncementState } from "./modules/announcement";
import apiPlugin from "./plugins/api";

export interface State {
  auth: AuthState;
  apiKeys: ApiKeysState;
  billing: NamespacesState;
  customer: CustomerState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol("store");

export const store = createStore<State>({
  modules: {
    auth,
    apiKeys,
    customer,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
  plugins: [
    apiPlugin,
  ],
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
