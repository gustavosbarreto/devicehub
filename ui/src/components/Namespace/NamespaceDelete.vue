<template>
  <v-tooltip location="bottom" class="text-center" :disabled="hasAuthorization">
    <template v-slot:activator="{ props }">
      <div v-bind="props">
        <v-btn
          :disabled="!hasAuthorization"
          color="red darken-1"
          variant="outlined"
          data-test="delete-btn"
          @click="dialog = !dialog"
        >
          Delete namespace
        </v-btn>
      </div>
    </template>
    <span> You don't have this kind of authorization. </span>
  </v-tooltip>

  <v-dialog v-model="dialog" max-width="540">
    <v-card data-test="namespaceDelete-dialog" class="bg-v-theme-surface">
      <v-card-title class="text-h5 pa-4 bg-primary">
        Are you sure?
      </v-card-title>

      <v-card-text class="mt-4 mb-3 pb-1">
        <p
          v-if="hasAuthorization && billingActive && billingInfo != undefined"
          data-test="contentSubscription-p"
        >
          Deleting the namespace will generate an invoice, estimated
          <b> {{ formatCurrency(billingInfo.nextPaymentDue, billingInfo.currency) }} </b> for the time
          of use.
        </p>

        <p data-test="content-text">
          This action cannot be undone. This will permanently delete the
          <b> {{ displayOnlyTenCharacters(name) }} </b>and its related data.
        </p>
      </v-card-text>

      <v-card-actions>
        <v-spacer />

        <v-btn variant="text" data-test="close-btn" @click="dialog = !dialog">
          Close
        </v-btn>

        <v-btn
          color="red darken-1"
          variant="text"
          data-test="remove-btn"
          @click="remove()"
        >
          Remove
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script lang="ts">
import { defineComponent, ref, computed, onMounted } from "vue";
import { useRouter } from "vue-router";
import { useStore } from "../../store";
import hasPermission from "../../utils/permission";
import { actions, authorizer } from "../../authorizer";
import { envVariables } from "../../envVariables";
import { displayOnlyTenCharacters } from "../../utils/string";
import formatCurrency from "@/utils/currency";
import {
  INotificationsError,
  INotificationsSuccess,
} from "../../interfaces/INotifications";

export default defineComponent({
  props: {
    nsTenant: {
      type: String,
      required: true,
    },
  },
  setup(props) {
    const store = useStore();
    const router = useRouter();
    const dialog = ref(false);
    const name = ref("");

    const tenant = computed(() => props.nsTenant);
    const billingActive = computed(() => store.getters["billing/active"]);
    const billing = computed(() => store.getters["billing/get"]);
    const billingInfo = computed(
      () => store.getters["billing/getBillInfoData"].info,
    );

    const hasAuthorization = computed(() => {
      const role = store.getters["auth/role"];
      if (role !== "") {
        return hasPermission(
          authorizer.role[role],
          actions.namespace.remove,
        );
      }
      return false;
    });

    const isBillingEnabled = () => envVariables.billingEnable;

    const getSubscriptionInfo = async () => {
      if (billingActive.value) {
        try {
          await store.dispatch("billing/getSubscription");
        } catch (error: any) {
          store.dispatch("snackbar/showSnackbarErrorDefault");
          throw new Error(error);
        }
      }
    };

    onMounted(() => {
      if (hasAuthorization.value && isBillingEnabled()) {
        getSubscriptionInfo();
      }

      name.value = store.getters["namespaces/get"].name;
    });

    const getDueAmount = (data: any) => data.upcoming_invoice.amount_due;

    const remove = async () => {
      try {
        dialog.value = !dialog.value;
        await store.dispatch("namespaces/remove", tenant.value);
        await store.dispatch("auth/logout");
        await store.dispatch("layout/setLayout", "simpleLayout");
        await router.push({ name: "login" });
        store.dispatch(
          "snackbar/showSnackbarSuccessAction",
          INotificationsSuccess.namespaceDelete,
        );
      } catch (error: any) {
        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.namespaceDelete,
        );
        throw new Error(error);
      }
    };

    return {
      dialog,
      hasAuthorization,
      name,
      tenant,
      billing,
      billingInfo,
      billingActive,
      isBillingEnabled,
      getSubscriptionInfo,
      getDueAmount,
      displayOnlyTenCharacters,
      formatCurrency,
      remove,
    };
  },
});
</script>
