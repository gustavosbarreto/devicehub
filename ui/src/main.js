import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
