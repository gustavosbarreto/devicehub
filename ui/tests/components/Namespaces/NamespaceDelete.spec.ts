import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import NamespaceDelete from "../../../src/components/Namespace/NamespaceDelete.vue";
import { createStore } from "vuex";
import { key } from "../../../src/store";
import routes from "../../../src/router";
import { envVariables } from "../../../src/envVariables";

const infoData = {
  info: {
    periodEnd: "2021-12-24T18:16:21Z",
    description: "Shellhub",
    latestPaymentDue: 0,
    latestPaymentPaid: 0,
    nextPaymentDue: 0,
    nextPaymenPaid: 0,
  },
  card: {
    brand: "visa",
    expYear: 2024,
    default: true,
    expMonth: 4,
    last4: "4042",
    id: "pm_1JzQ80KJsksFHO6pREJA5TrK",
  },
  cards: [
    {
      brand: "visa",
      expYear: 2024,
      default: true,
      expMonth: 4,
      last4: "4042",
      id: "pm_1JzQ80KJsksFHO6pREJA5TrK",
    },
    {
      brand: "visa",
      expYear: 2028,
      default: false,
      expMonth: 4,
      last4: "4042",
      id: "pm_1JzQ80KJsksFHO6pREJA5TrG",
    },
    {
      brand: "visa",
      expYear: 2029,
      default: false,
      expMonth: 4,
      last4: "4042",
      id: "pm_1JzQ80KJsksFHO6pREJA5TrF",
    },
  ],
};

const nsTenant = "xxxxxx";

const role = ["owner", "operator"];

const hasAuthorization = {
  owner: true,
  operator: false,
};

const inactiveBilling = {
  active: false,
  current_period_end: 0,
  customer_id: "",
  subscription_id: "",
  payment_method_id: "",
};

const activeBilling = {
  active: true,
  current_period_end: 12121,
  customer_id: "cus_123",
  subscription_id: "subs_123",
  payment_method_id: "pm_123",
};

const namespaceObject = {
  name: "namespace3",
  owner: "user1",
  member_names: ["user1", "user7", "user8"],
  tenant_id: "xxxxxxxx",
};

const getter = {
  billingActive: [inactiveBilling, activeBilling],
};

const tests = [
  {
    description: "Button",
    props: {
      nsTenant,
    },
    data: {
      name: namespaceObject.name,
      dialog: false,
      action: "remove",
    },
    computed: {
      tenant: nsTenant,
      active: getter.billingActive[0].active,
      billing: getter.billingActive[0],
      billingActive: getter.billingActive[0].active,
    },
    namespace: namespaceObject,
    env: {
      billingEnable: false,
    },
    info: {},
    template: {
      "delete-btn": true,
      "namespaceDelete-dialog": false,
      "contentSubscription-p": false,
      "close-btn": false,
      "remove-btn": false,
    },
  },
  {
    description: "Dialog without subscription",
    props: {
      nsTenant,
    },
    data: {
      name: namespaceObject.name,
      dialog: false,
      action: "remove",
    },
    computed: {
      tenant: nsTenant,
      active: true,
      billing: getter.billingActive[0],
      billingActive: true,
    },
    namespace: namespaceObject,
    env: {
      billingEnable: false,
    },
    info: {},
    template: {
      "delete-btn": true,
      "namespaceDelete-dialog": true,
      "contentSubscription-p": false,
      "close-btn": true,
      "remove-btn": true,
    },
  },
  {
    description: "Dialog with subscription",
    props: {
      nsTenant,
    },
    data: {
      name: namespaceObject.name,
      dialog: false,
      action: "remove",
    },
    computed: {
      tenant: nsTenant,
      active: false,
      billing: getter.billingActive[1],
      billingActive: false,
    },
    namespace: namespaceObject,
    env: {
      billingEnable: true,
    },
    info: infoData,
    template: {
      "delete-btn": true,
      "namespaceDelete-dialog": true,
      "contentSubscription-p": true,
      "close-btn": true,
      "remove-btn": true,
    },
  },
];

const store = (
  active: boolean,
  billing: any,
  currentrole: any,
  namespace: any,
  info: any
) => {
  return createStore({
    state: {
      active,
      billing,
      currentrole,
      namespace,
      info,
    },
    getters: {
      "billing/active": (state) => state.active,
      "billing/get": (state) => state.billing,
      "auth/role": (state) => state.currentrole,
      "namespaces/get": (state) => state.namespace,
      "billing/getBillInfoData": (state) => state.info,
    },
    actions: {
      "namespaces/remove": vi.fn(),
      "auth/logout": vi.fn(),
      "billing/getSubscription": vi.fn(),
      "snackbar/showSnackbarSuccessAction": vi.fn(),
      "snackbar/showSnackbarErrorAction": vi.fn(),
      "snackbar/showSnackbarErrorDefault": vi.fn(),
    },
  });
};

describe("NamespaceDelete", () => {
  let wrapper: VueWrapper<any>;
  const vuetify = createVuetify();

  tests.forEach((test) => {
    role.forEach((currentrole) => {
      describe(`${test.description} ${currentrole}`, () => {
        beforeEach(() => {
          wrapper = mount(NamespaceDelete, {
            global: {
              plugins: [
                [
                  store(
                    test.computed.active,
                    test.computed.billing,
                    currentrole,
                    test.namespace,
                    test.info
                  ),
                  key,
                ],
                routes,
                vuetify,
              ],
            },
            props: { nsTenant: test.props.nsTenant },
            mocks: {
              $stripe: {
                elements: () => ({
                  create: () => ({
                    mount: () => null,
                  }),
                }),
              },
            },
            shallow: true,
          });

          envVariables.billingEnable = test.env.billingEnable;
        });

        ///////
        // Component Rendering
        //////

        it("Is a Vue instance", () => {
          expect(wrapper).toBeTruthy();
        });
        it("Renders the component", () => {
          expect(wrapper.html()).toMatchSnapshot();
        });

        ///////
        // Data checking
        //////
        it("Data is defined", () => {
          expect(wrapper.vm.$data).toBeDefined();
        });

        it("Receive data in props", () => {
          Object.keys(test.props).forEach((item) => {
            // @ts-ignore
            expect(wrapper.vm[item]).toEqual(test.props[item]);
          });
        });

        it("Receive data in data", () => {
          expect(wrapper.vm.dialog).toEqual(test.data.dialog);
          expect(wrapper.vm.name).toEqual(test.data.name);
        });
        it("Process data in the computed", () => {
          expect(wrapper.vm.tenant).toEqual(test.computed.tenant);
          expect(wrapper.vm.billingActive).toEqual(test.computed.billingActive);
          expect(wrapper.vm.billing).toEqual(test.computed.billing);
        });

        //////
        // HTML validation
        //////
        // TODO
      });
    });
  });
});
