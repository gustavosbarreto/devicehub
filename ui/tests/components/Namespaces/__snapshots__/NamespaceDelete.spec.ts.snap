// Vitest Snapshot v1

exports[`NamespaceDelete > Button operator > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;

exports[`NamespaceDelete > Button owner > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"true\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;

exports[`NamespaceDelete > Dialog with subscription operator > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;

exports[`NamespaceDelete > Dialog with subscription owner > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"true\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;

exports[`NamespaceDelete > Dialog without subscription operator > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;

exports[`NamespaceDelete > Dialog without subscription owner > Renders the component 1`] = `
"<v-tooltip-stub attach=\\"false\\" closeonback=\\"false\\" contained=\\"false\\" disabled=\\"true\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" scrim=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonclick=\\"false\\" openonhover=\\"true\\" closeoncontentclick=\\"false\\" minwidth=\\"0\\" locationstrategy=\\"connected\\" location=\\"bottom\\" origin=\\"auto\\" offset=\\"10\\" scrollstrategy=\\"reposition\\" transition=\\"false\\" class=\\"text-center\\"></v-tooltip-stub>
<v-dialog-stub maxwidth=\\"540\\" fullscreen=\\"false\\" retainfocus=\\"true\\" scrollable=\\"false\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" noclickanimation=\\"false\\" modelvalue=\\"false\\" persistent=\\"false\\" scrim=\\"true\\" zindex=\\"2400\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" location=\\"bottom\\" origin=\\"center center\\" scrollstrategy=\\"block\\" transition=\\"[object Object]\\"></v-dialog-stub>"
`;
